tests/test_utils.py

```diff
@@ -60,3 +65,120 @@ def test_to_absolute_path_without_hydra(
     path = str(Path(path))
     expected = str(Path(expected).absolute())
     assert utils.to_absolute_path(path) == expected
+
+
+class TestRunAndReport:
+    class DemoFunctions:
+        """Demo inputs for `run_and_report`"""
```

Comment:
Can you explain the testing logic?
The test is surprisingly large.

Reply:
> The test is surprisingly large.

Sorry about that.
This is not just testing the new functionality. It is also testing the pre-existing `run_and_report` behavior.

The `run_and_report` function has the following signature:
```
def run_and_report(func: Any) -> Any: ...
```

There are three test methods:
- `test_success` tests when `run_and_report` runs `func` successfully.
- `test_failure` tests when `func` raises an exception and `run_and_report` is able to print a nicely-formatted error message.
- `test_simplified_traceback_failure` tests when `func` raises an exception and `run_and_report` is *not* able to print a nicely-formatted error message -- this is the case where the original exception gets re-raised.

There are four `@staticmethod` definitions. These are passed in to `run_and_report` as the `func` argument. They are designed to trigger different logic code paths inside of `run_and_report`:
- `success_func` raises no exception
- `simple_error` raises an `AssertionError`
- `run_job_wrapper` triggers the special logic in `run_and_report` that looks for a function called `"run_job"` in the stack, and strips away the leading stack frames
- `omegaconf_job_wrapper` triggers the special logic in `run_and_report` that looks for the `omegaconf` module in the stack, and strips away the bottom stack frames

Reply:
Gotcha.
Thanks for adding this. Can you add a comment describing this in the test class?
(does not need to be as detailed).